src/Sentry/Protocol/Envelopes/EnvelopeItem.cs

```diff
@@ -176,6 +177,19 @@ public static EnvelopeItem FromUserFeedback(UserFeedback sentryUserFeedback)
             return new EnvelopeItem(header, new JsonSerializable(sentryUserFeedback));
         }
 
+        /// <summary>
+        /// Creates an envelope item from transaction.
+        /// </summary>
+        public static EnvelopeItem FromTransaction(Transaction transaction)
+        {
+            var header = new Dictionary<string, object?>(StringComparer.Ordinal)
```

Comment:
Does this change? Can we get away with a single instance that we pass the reference of to all envelope items?

Reply:
Technically it should also have `event_id` but it's not currently exposed on `Transaction` and it works without it. Not sure if I should actually try to get that in.

Reply:
Maybe that should be documented as a TODO now, if it is not yet included but could be added to the implementation later on?

Reply:
True. Also if it's not needed then maybe we can get away with it. @jan-auer do we need the `event_id` in the envelope header afterall? it's working without it and we could avoid allocating a new map just to set the id (which is anyway on the payload)